Sources/HTTP/HTTPHeadersAccessors.swift

```diff
@@ -0,0 +1,1596 @@
+            let segments = text.components(separatedBy: "=")
+            guard let key = segments.first?.trimmingCharacters(in: .whitespaces), !key.isEmpty else {
+                return nil
+            }
+            let value = segments.dropFirst().joined(separator: "=")
+            return (key, value)
+            }) ?? []
+        
+        return pairs.flatMap {
+            // path should be set otherwise it will fail!
+            return HTTPCookie(properties: [.name : $0.key, .value: $0.val, .path: "/"])
+        }
+    }
+    
+    /// `If-Match` header etag value.
+    public var ifMatch: [EntryTag] {
+        get {
+            return (self.storage[.ifMatch] ?? []).map(EntryTag.init)
+        }
+        set {
+            // TOCHECK: When there is a wildcard, other values should be ignored
+            if !newValue.isEmpty {
+                self.storage[.ifMatch] = newValue.map { $0.description }
+            } else {
+                self.storage[.ifMatch] = nil
+            }
+        }
+    }
+    
+    /// `If-None-Match` header etag value.
+    public var ifNoneMatch: [EntryTag] {
+        get {
+            return (self.storage[.ifNoneMatch] ?? []).map(EntryTag.init)
+        }
+        set {
+            // TOCHECK: When there is a wildcard, other values should be ignored
+            if !newValue.isEmpty {
+                self.storage[.ifNoneMatch] = newValue.map { $0.description }
+            } else {
+                self.storage[.ifNoneMatch] = nil
+            }
+        }
+    }
+    
+    /// `If-Range` header etag value.
+    public var ifRange: HTTPHeaders.IfRange? {
+        get {
+            return self.storage[.ifRange]?.first.flatMap(IfRange.init)
+        }
+        set {
+            self.storage[.ifRange] = newValue.flatMap { [$0.description] }
+        }
+    }
+    
+    /// `If-Modified-Since` header value.
+    public var ifModifiedSince: Date? {
+        get {
+            return self.storage[.ifModifiedSince]?.first.flatMap(Date.init(rfcString:))
+        }
+        set {
+            self.storage[.ifModifiedSince] = newValue.flatMap { [$0.format(with: .http)] }
+        }
+    }
+    
+    /// `If-Unmodified-Since` header value.
+    public var ifUnmodifiedSince: Date? {
+        get {
+            return self.storage[.ifUnmodifiedSince]?.first.flatMap(Date.init(rfcString:))
+        }
+        set {
+            self.storage[.ifUnmodifiedSince] = newValue.flatMap { [$0.format(with: .http)] }
+        }
+    }
+    
+    /// `Origin` header value.
+    public var origin: URL? {
+        get {
+            return self.storage[.origin]?.first.flatMap(URL.init(string:))
+        }
+        set {
+            self.storage[.origin] = newValue.flatMap { [$0.absoluteString] }
+        }
+    }
+    
+    /// Returns `Range` header value.
+    /// - Note: upperbound will be Int64.max in case of open ended Range.
+    public var range: Range<Int64>? {
```

Comment:
According to RFC7233 range headers can have multiple headers. So this should be defined as `[Range<Int64>]?`.

Also the spec for parsing ranges is not as simple as below 4 lines but I guess that can be addressed later :p

Reply:
As far as I understood from examples in RFC, multiple ranges are continuous if we intersect values. e.g:

`bytes=0-0,-1` becomes: 0..<2
`bytes=500-600,601-999` becomes: 500..<1000
`bytes=500-700,601-999` becomes: 500..<1000

Thus it is not an array of range. Please correct me if I'm wrong?

Reply:
Spec defines various ranges, they do not have to be continuous, for example response in page 11 shows at least two ranges bytes=500-999,7000-7999

As per spec, server CAN merge overlapping or adjacent ranges however that is not a must. Multipart response for partial content is clearly defined which indicates the existence of multiple ranges. 

     HTTP/1.1 206 Partial Content
     Date: Wed, 15 Nov 1995 06:25:24 GMT
     Last-Modified: Wed, 15 Nov 1995 04:58:08 GMT
     Content-Length: 1741
     Content-Type: multipart/byteranges; boundary=THIS_STRING_SEPARATES

     --THIS_STRING_SEPARATES
     Content-Type: application/pdf
     Content-Range: bytes 500-999/8000

     ...the first range...
     --THIS_STRING_SEPARATES
     Content-Type: application/pdf
     Content-Range: bytes 7000-7999/8000

     ...the second range
     --THIS_STRING_SEPARATES--


https://tools.ietf.org/html/rfc7233

Reply:
Spec defines various ranges, they do not have to be continuous, for example response in page 11 shows at least two ranges bytes=500-999,7000-7999

As per spec, server CAN merge overlapping or adjacent ranges however that is not a must. Multipart response for partial content is clearly defined which indicates the existence of multiple ranges. 

     HTTP/1.1 206 Partial Content
     Date: Wed, 15 Nov 1995 06:25:24 GMT
     Last-Modified: Wed, 15 Nov 1995 04:58:08 GMT
     Content-Length: 1741
     Content-Type: multipart/byteranges; boundary=THIS_STRING_SEPARATES

     --THIS_STRING_SEPARATES
     Content-Type: application/pdf
     Content-Range: bytes 500-999/8000

     ...the first range...
     --THIS_STRING_SEPARATES
     Content-Type: application/pdf
     Content-Range: bytes 7000-7999/8000

     ...the second range
     --THIS_STRING_SEPARATES--


https://tools.ietf.org/html/rfc7233